node/coinstacks/solana/api/src/websocket.ts

```diff
@@ -0,0 +1,117 @@
+import { Logger } from '@shapeshiftoss/logger'
+import { WebsocketClient as BaseWebsocketClient, Args, Options, Subscription } from '@shapeshiftoss/websocket'
+import WebSocket from 'ws'
+import {
+  isWebsocketResponse,
+  isWebsocketSubscribeResponse,
+  WebsocketResponse,
+  WebsocketSubscribeResponse,
+} from './types'
+import { Logs } from '@solana/web3.js'
+
+const logger = new Logger({ namespace: ['unchained', 'solana', 'websocket'], level: process.env.LOG_LEVEL })
+
+interface WebsocketArgs extends Omit<Args, 'logger'> {
+  transactionHandler: TransactionHandler | Array<TransactionHandler>
+}
+
+type TransactionHandler = (data: Logs) => Promise<void>
+
+export class WebsocketClient extends BaseWebsocketClient {
+  private handleTransaction: TransactionHandler | Array<TransactionHandler>
+  private addresses: Array<string> = []
+  private subscriptionsIds: Map<number, number> = new Map()
```

Comment:
doesn't feel like the map is actually useful, seems like a simple number array would be sufficient

Reply:
The map is used to match the subscriptionId with the incremental id, we need a mapping of number => subId

We should keep it or use an array of tuples, I'm agnostic

Reply:
I guess my thought was if we are unsubscribing from all tracked subscriptions on `subscribeAddresses` and only storing `subscriptionIds` if they match the current incremental id (or unsubscribing immediately), then just a filter on the current list of subscription ids would be sufficient at message handle time. Not a huge deal either way, just a thought.